File: pfce/pfce_src/pro/Web/public/ts/app/models/Library.ts
/// <reference path="../reference.ts" />

namespace Peach {
	export interface ILibrary {
		libraryUrl: string;
		name: string;
		description: string;
		locked: boolean;
		versions: ILibraryVersion[];
		groups: IGroup[];
		user: string;
		timeStamp: Date;
	}

	export interface ILibraryVersion {
		version: number;
		locked: boolean;
		pits: IPit[];
	}

	export interface IGroup {
		groupUrl: string;
		access: string;
	}

	export interface ITag {
		name: string;
		values: string[];
	}

	export interface IError {
		errorMessage: string;
	}
}

File: pfce/pfce_src/pro/Core/WebServices/Models/Library.cs
using System;
using System.Collections.Generic;

namespace Peach.Pro.Core.WebServices.Models
{
	public class Library
	{
		public string LibraryUrl { get; set; }
		public string Name { get; set; }
		public string Description { get; set; }

		public bool Locked { get; set; }

		public List<LibraryVersion> Versions { get; set; }

		public List<Group> Groups { get; set; }

		public string User { get; set; }

		public DateTime Timestamp { get; set; }
	}
}
